pkg/meta/redis.go

```diff
@@ -4240,3 +4247,60 @@ func (m *redisMeta) doAttachDirNode(ctx Context, parent Ino, dstIno Ino, name st
 		return err
 	}, m.inodeKey(parent), m.entryKey(parent)))
 }
+
+// caller makes sure inode is not special inode.
+func (m *redisMeta) touchAtime(ctx Context, ino Ino, cur *Attr) (rerr syscall.Errno) {
+	if (m.conf.AtimeMode != StrictAtime && m.conf.AtimeMode != RelAtime) || m.conf.ReadOnly {
+		return 0
+	}
+
+	var attr *Attr
+	newAttr := &Attr{}
+	if cur != nil {
+		attr = cur
+	} else if m.of.Check(ino, newAttr) {
+		attr = newAttr
+	}
+
+	updated := false
+	defer func() {
+		if rerr == 0 && m.of.IsOpen(ino) && updated {
+			m.of.Update(ino, attr)
+		}
+	}()
+
+	now := time.Now()
+
+	// Already got attr, update atime without transaction
+	if attr != nil {
+		if !m.atimeNeedsUpdate(attr, now) {
+			return 0
+		}
+		attr.Atime = now.Unix()
+		attr.Atimensec = uint32(now.Nanosecond())
+		updated = true
+		return errno(m.rdb.Set(ctx, m.inodeKey(ino), m.marshal(attr), 0).Err())
```

Comment:
We can't call rdb.Set without a transaction, because it may overwrite the change in attr from others.

Reply:
@SandyXSD suggested to avoid transaction when possible, so you mean only redis always requires transactions? tkv and sql may still avoid transactions when possible?

Reply:
We can't only update the atime of Attr, so we can't avoid the transaction.

Reply:
Fixed, always update atime in transaction, this basically reverts "meta: avoid transaction if possible in touchAtime()".

Reply:
My bad. The actual expected behavior should be: avoid updating whenever possible.
In other words, the procedure is like:
1. if we already have the attr in memory, use it to decide whether the atime should be updated
2. if not cached, or the atime SHOULD be updated, use a transaction to update the atime
3. and in the transaction, we should get the latest attr first, and update only the atime if needed